dotnetv3/S3/scenarios/TransferUtilityBasics/TransferUtilityBasicsTests/TransferMethodsTests.cs

```diff
@@ -1,65 +1,87 @@
+// SPDX-License-Identifier:  Apache-2.0
 
 namespace TransferUtilityBasics.Tests
 {
-    [TestClass()]
     public class TransferMethodsTests
     {
-        IAmazonS3 client;
-        TransferUtility transferUtil;
-        string LocalPath = $"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\TransferFolderTest";
-        string bucketName;
+        private readonly IConfiguration _configuration;
 
-        TransferMethodsTests()
+        readonly IAmazonS3 _client;
+        readonly TransferUtility transferUtil;
+        private readonly string _localPath = $"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\TransferFolderTest";
+
+        public TransferMethodsTests()
         {
-            client = new AmazonS3Client();
-            transferUtil = new TransferUtility();
-            bucketName = "doc-example-bucket1-test";
+            _client = new AmazonS3Client();
+            transferUtil = new TransferUtility(_client);
+
+            _client = new AmazonS3Client();
+            _configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("testsettings.json") // Load test settings from JSON file.
+                .AddJsonFile("testsettings.local.json",
+                    true) // Optionally load local settings.
+                .Build();
         }
 
-        [TestMethod()]
+        [Fact()]
         public async Task DownloadSingleFileAsyncTest()
         {
-            var keyName = "FileToDownload.docx";
-            var success = await TransferMethods.DownloadSingleFileAsync(transferUtil, bucketName, keyName, LocalPath);
-            Assert.IsTrue(success, $"Couldn't download {keyName}.");
+            var keyName = _configuration["FileToDownload"];
+            var success = await TransferMethods.DownloadSingleFileAsync(
+                transferUtil,
+                _configuration["BucketName"],
+                _configuration["FileToDownload"],
+                _localPath);
+
+            Assert.True(success, $"Couldn't download {keyName}.");
         }
 
-        [TestMethod()]
+        [Fact()]
         public async Task DownloadS3DirectoryAsyncTest()
         {
-            var downloadPath = $"{LocalPath}\\TestDownloadFolder";
-            var s3Path = "DownloadTest";
+            var downloadPath = $"{_localPath}\\TestDownloadFolder";
+            var s3Path = _configuration["S3Path"];
+
+            var success = await TransferMethods.DownloadS3DirectoryAsync(
+                transferUtil,
+                _configuration["BucketName"],
+                downloadPath,
+                s3Path);
 
-            var success = await TransferMethods.DownloadS3DirectoryAsync(transferUtil, bucketName, downloadPath, s3Path);
-            Assert.IsTrue(success, "Couldn't download files from {s3Path}.");
+            Assert.True(success, "Couldn't download files from {s3Path}.");
```

Comment:
Same question as above. Can you make sure the directory is empty or does not exist before the test, and it is not empty after?

Reply:
I've modified the code to make that test and properly return true or false based on whether any files were downloaded. It would be redundant to do it here as well, I think.